docs/readmes/resources/ref_magma_metrics.md

```diff
@@ -0,0 +1,186 @@
+---
+id: ref_magma_metrics
+title: Life of a Magma Metric
+hide_title: true
+---
+
+## Life of a Magma Metric
+
+This document will cover the life of a Magma metric as it originates at the
+access gateway (AGW), goes through the orchestrator (orc8r) to the [Prometheus
+Edge Hub](https://github.com/facebookincubator/prometheus-edge-hub), and eventually gets displayed on the NMS UI.
+
+### Phase 1: Collection and Export
+
+On the AGW, metrics are collected and exported from either Python services or
+C++ services.
+
+Every **Python** service on the access gateway defines
+[Prometheus](https://prometheus.io/) Counters, Gauges, and Histograms in a
+`metrics.py` file. The `subscriberdb` service, for example, defines the
+following Prometheus metrics in its
+[metrics.py](https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/python/magma/subscriberdb/metrics.py)
+file among others:
+
+```python
+SUBSCRIBER_SYNC_SUCCESS_TOTAL = Counter(
+    'subscriber_sync_success',
+    'Total number of successful subscriber'
+    'syncs with cloud',
+)
+
+SUBSCRIBER_SYNC_FAILURE_TOTAL = Counter(
+    'subscriber_sync_failure',
+    'Total number of failed subscriber'
+    'syncs with cloud',
+)
+
+```
+
+It is then up to the service to set the values of these metrics appropriately,
+i.e., take the actual measurements. `subscriberdb` increments its
+`SUBSCRIBER_SYNC_SUCCESS_TOTAL` counter in its
+[client.py](https://sourcegraph.com/github.com/magma/magma/-/blob/lte/gateway/python/magma/subscriberdb/client.py)
+file after fetching subscriber data from the cloud successfully .
+
+```python
+logging.info(
+    "Successfully fetched all subscriber "
+    "pages from the cloud!",
+)
+SUBSCRIBER_SYNC_SUCCESS_TOTAL.inc()
+```
+
+Each Python service on the AGW extends the `MagmaService` class which implements
+the Service303 interface. This interface has the
+[`GetMetrics`](https://sourcegraph.com/github.com/magma/magma/-/blob/orc8r/gateway/python/magma/common/service.py?L403:9)
```

Comment:
Throughout: link your refs to a particular git tag (e.g. v1.6, or a particular commit) because otherwise the line numbers can and will change over time and will cause this documentation to go out of date much quicker